blueocean-dashboard/src/main/js/OrganizationPipelines.jsx

```diff
@@ -38,11 +34,7 @@ class OrganizationPipelines extends Component {
         if (config) {
             const { organization } = this.context.params;
             this.props.fetchPipelinesIfNeeded(this.context.config, organization);
-            if (this.props.params.pipeline) {
-                const { pipeline } = this.props.params;
-                config.pipeline = pipeline;
-                this.props.setPipeline(config);
-            }
+
```

Comment:
The above call to `setPipeline` was triggering a second fetch for pipelines - but only if the user deep-linked to a route with the :pipeline route param. These two calls were causing odd behavior depending on which one completed last. I decided to remove this, and instead let `componentWillReceiveProps()` execute this logic each time. This eliminates the timing issue and ensures the right pipeline is "selected" any time the props change.